src/components/ProjectSettingsRedirect.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from "@apollo/client";
+import { Redirect } from "react-router-dom";
+import {
+  getProjectSettingsRoute,
+  ProjectSettingsTabRoutes,
+} from "constants/routes";
+import { GetViewableProjectRefsQuery } from "gql/generated/types";
+import { GET_VIEWABLE_PROJECTS } from "gql/queries";
+
+export const ProjectSettingsRedirect: React.FC = () => {
+  const { data } = useQuery<GetViewableProjectRefsQuery>(GET_VIEWABLE_PROJECTS);
+
+  if (data) {
+    const { viewableProjectRefs } = data;
+    if (
+      viewableProjectRefs.length !== 0 &&
+      viewableProjectRefs[0]?.projects.length !== 0
+    ) {
+      return (
+        <Redirect
+          to={getProjectSettingsRoute(
+            data.viewableProjectRefs[0]?.projects[0]?.identifier,
```

Comment:
Since we know these fields exist at this point. 
```suggestion
            viewableProjectRefs[0].projects[0].identifier,
```